lib/collection/proxy-config.js

```diff
@@ -97,13 +122,37 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
      * @param {String=} [urlStr] The url string for which the proxy match needs to be done.
      */
     test: function(urlStr) {
-        // this ensures we don't proceed any further for any protocol
-        // that is not http or https
-        if (_.isEmpty(urlStr) || (_.isString(urlStr) && !urlStr.startsWith('http'))) {
```

Comment:
I thought regex based approach would be better for future when we support ftp or socks